firedrake/utility_meshes.py

```diff
@@ -899,3 +899,90 @@ def TorusMesh(nR, nr, R, r, quadrilateral=False, reorder=None):
     plex = mesh._from_cell_list(2, cells, vertices)
     m = mesh.Mesh(plex, dim=3, reorder=reorder)
     return m
+
+
+@profile
+def CylinderMesh(nr, nz, radius=1, depth=1, quadrilateral=False, reorder=None):
+    """Generates a cylinder mesh.
+
+    :arg nr: number of cells the cylinder circumference should be
+         divided into (min 3)
+    :arg nz: number of cells along the longitudinal z-axis of the cylinder
+    :kwarg radius: (optional) radius of the cylinder to approximate
+         (default 1).
+    :kwarg quadrilateral: (optional), creates quadrilateral mesh, defaults to False
+    """
+    if nr < 3:
+        raise ValueError("CircleManifoldMesh must have at least three cells")
+
+    coord_xy = radius*np.column_stack((np.cos(np.arange(nr)*(2*np.pi/nr)),
+                                       np.sin(np.arange(nr)*(2*np.pi/nr))))
+    coord_z = depth*np.linspace(0.0, 1.0, nz + 1).reshape(-1, 1)
+    vertices = np.column_stack((np.tile(coord_xy, (nz + 1, 1)),
+                                np.tile(coord_z, (1, nr)).reshape(-1, 1)))
+
+    # intervals on circumference
+    ring_cells = np.column_stack((np.arange(0, nr, dtype=np.int32),
+                                  np.roll(np.arange(0, nr, dtype=np.int32), -1)))
+    # quads in the first layer
+    ring_cells = np.column_stack((ring_cells, np.roll(ring_cells, 1, axis=1) + nr))
+    scalar = np.arange(nz)*nr
+    cells = np.row_stack((ring_cells + i for i in scalar))
+    if not quadrilateral:
+        # two cells per cell above...
+        cells = cells[:, [0, 1, 3, 1, 2, 3]].reshape(-1, 3)
+
+    plex = mesh._from_cell_list(2, cells, vertices)
+    m = mesh.Mesh(plex, dim=3, reorder=reorder)
+    return m
+
+
+def PeriodicRectangleMeshX(nx, ny, Lx, Ly, quadrilateral=False, reorder=None):
```

Comment:
Eventually this could be incorporated in PeriodicRectangleMesh via keyword argument `periodicity` that could be `'x'`, `'y'`, or `'both'`?